src/actions.cpp

```diff
@@ -531,6 +602,19 @@ void CAction::setPcapArgs (pcap_pkts  *  P_value)
 
 void CAction::setPcapArgs(const char* P_value)
 {
+    if (P_value[0] == '~')
+    {
+        char pcap_absolute_path[MAX_PATH] = {'\0'};
+        if (expand_user_path(P_value, pcap_absolute_path))
+        {
+            P_value = strndup(pcap_absolute_path, strlen(pcap_absolute_path));
+        }
+        else
+        {
+            ERROR("Couldn't resolve pcap file absolute path: %s", P_value);
+        }
+    }
+
```

Comment:
This would make a lot more sense to me:
```diff
--- a/src/send_packets.c
+++ b/src/send_packets.c
@@ -95,13 +95,20 @@ float2timer(float time, struct timeval *tvp)
 
 static char* find_file(const char* filename)
 {
+    char tmppath[MAX_PATH];
     char *fullpath;
-    if (filename[0] == '/' || !*scenario_path) {
-        return strdup(filename);
+    size_t len;
+
+    tmppath[0] = '\0';
+    expand_user_path(filename, tmppath, sizeof(tmppath));
+
+    if (tmppath[0] == '/' || !*scenario_path) {
+        return strdup(tmppath);
     }
 
-    fullpath = malloc(MAX_PATH);
-    snprintf(fullpath, MAX_PATH, "%s%s", scenario_path, filename);
+    len = strlen(scenario_path) + strlen(tmppath) + 1;
+    fullpath = malloc(len);
+    snprintf(fullpath, len, "%s%s", scenario_path, tmppath);
 
     if (access(fullpath, R_OK) < 0) {
         free(fullpath);
```
If you start from there, I think the rest will fall into place.